pkg/env/prod.go

```diff
@@ -139,6 +141,20 @@ func newProd(ctx context.Context, log *logrus.Entry) (*prod, error) {
 
 	p.serviceKeyvault = keyvault.NewManager(msiKVAuthorizer, serviceKeyvaultURI)
 
+	client := mgmtclusters.NewManagedClustersClient(p.SubscriptionID())
```

Comment:
If we are going this way - we should probably have a wrapper in `pkg/util/azureclient` with an interface so we can replace this with a mock.